.github/workflows/build.yml

```diff
@@ -150,15 +150,17 @@ jobs:
           - name: linux-O0
             os: ubuntu-latest
             config_arg: CFLAGS='-O0'
-          - name: macos
-            os: macos-latest
+          - name: macos-x86_64
+            os: macos-13
+          - name: macos-arm64
+            os: macos-14
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           persist-credentials: false
       - name: OS Dependencies
-        if: runner.os == 'MacOS'
+        if: ${{ startsWith(matrix.os, 'macos') }}
```

Comment:
The string for `runner.os` I think is actually `macOS` (although I presume it doesn't matter), but otherwise what's the reason for this change?

Reply:
I thought it was more obvious to base this on the `matrix.os` value rather than `runner.os`, since we set that a few lines above. `startsWith` is [not case sensitive](https://docs.github.com/en/actions/learn-github-actions/expressions#startswith).
However it works without this change so it can be dropped.

Reply:
I think the fact we need to install dependencies is a property of the OS it's physically running on, not what we declare in the matrix (so I'd prefer to keep it based on `runner.os`)

Reply:
Make sense I've removed that change.